File: vanilla-dashboard/temp_phase_check.js
const fs = require('fs');
const manifest = JSON.parse(fs.readFileSync('data/Patient_12738_OP/notes_manifest.json', 'utf8'));
const admission = new Date(2023, 4, 4);
const discharge = new Date(2023, 4, 11);
const dayMs = 24 * 60 * 60 * 1000;
function mapDateToPhase(noteDate, admission, discharge) {
  if (!noteDate || !admission || !discharge) return null;
  const n = new Date(noteDate);
  n.setHours(0, 0, 0, 0);
  const a = new Date(admission);
  const d = new Date(discharge);
  a.setHours(0, 0, 0, 0);
  d.setHours(0, 0, 0, 0);
  if (n < new Date(a.getTime() - dayMs)) return null;
  if (n <= a) return 'Home';
  const erDate = new Date(a.getTime() + dayMs);
  if (n.getTime() === erDate.getTime()) return 'ER';
  if (n > erDate && n < d) return 'Unit';
  if (n.getTime() === d.getTime()) return 'Discharge';
  if (n <= new Date(d.getTime() + dayMs * 7)) return 'Post-Discharge';
  return null;
}
const grouped = { Home: [], ER: [], Unit: [], Discharge: [], 'Post-Discharge': [] };
for (const file of manifest.files) {
  const d = new Date(file.date);
  const phase = mapDateToPhase(d, admission, discharge);
  if (phase) grouped[phase].push(file.date);
}
for (const [phase, dates] of Object.entries(grouped)) {
  console.log(phase + ': ' + dates.join(', '));
}

File: vanilla-dashboard/temp_phase_check.py
import json
from datetime import date, datetime, timedelta
from pathlib import Path

manifest = json.loads(Path('data/Patient_12738_OP/notes_manifest.json').read_text(encoding='utf-8'))
admission = date(2023, 5, 4)
discharge = date(2023, 5, 11)
day = timedelta(days=1)

def map_date_to_phase(note_date, admission, discharge):
    if not note_date or not admission or not discharge:
        return None
    n = note_date
    a = admission
    d = discharge
    if n < a - day:
        return None
    if n <= a:
        return 'Home'
    er_date = a + day
    if n == er_date:
        return 'ER'
    if a < n < d:
        return 'Unit'
    if n == d:
        return 'Discharge'
    if n <= d + timedelta(days=7):
        return 'Post-Discharge'
    return None

out = {'Home': [], 'ER': [], 'Unit': [], 'Discharge': [], 'Post-Discharge': []}
for file in manifest['files']:
    d = datetime.strptime(file['date'], '%Y-%m-%d').date()
    phase = map_date_to_phase(d, admission, discharge)
    if phase:
        out[phase].append(file['date'])

for phase, dates in out.items():
    print(phase + ': ' + ', '.join(dates))
